tests/unit/AgmLocation.spec.ts

```diff
@@ -0,0 +1,102 @@
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+import { mount } from '@vue/test-utils'
+import AgmLocation from '@/components/AgmLocationChange/AgmLocation.vue'
+
+Vue.use(Vuetify)
+const vuetify = new Vuetify({})
+
+// Prevent the warning "[Vuetify] Unable to locate target [data-app]"
+document.body.setAttribute('data-app', 'true')
+
+describe('AgmLocation', () => {
+  /** Array of validations rules for AGM location to be passed as a prop. */
+  const rules = [] as Array<(val) => boolean | string>
+  rules.push(val => !!val || 'AGM location is required.')
+  rules.push(val => (val.length <= 100) || 'Must be 100 characters or less.')
+
+  it('initializes correctly', () => {
+    const wrapper = mount(AgmLocation,
+      {
+        vuetify,
+        propsData: {
+          rules: rules
+        }
+      })
+    const vm: any = wrapper.vm
+
+    expect(vm.agmLocation).toBe('')
+    expect(wrapper.emitted('update:agmLocation')).toBeUndefined()
+    expect(wrapper.emitted('valid').pop()[0]).toBe(false)
```

Comment:
How many "valid" events were emitted, here and below?

Reply:
What do you mean Sev? It's only one event which is an array of false and an empty string.
![valid events](https://github.com/bcgov/business-filings-ui/assets/122301442/f1e480ee-c7a4-49f0-8105-88bab4c4b3d5)
The first value is the emitted validity.

Reply:
In this test, "valid" is only emitted once. But in other tests it might be emitted 100 times...

Reply:
I just checked Sev, the others emit two. One for the onCreated and the other is what I'm trying to do (which is to be expected). For example:
![array test](https://github.com/bcgov/business-filings-ui/assets/122301442/5030a6d6-38d9-4cb8-8ec9-dcbba91f2f6c)

Reply:
I bet if you entered each "a" like a user typing, you'd have a lot more valid events (esp without the debouncing).

This is fine. Just remember to have a look at the number of events emitted in scenarios like there. "Popping" the last one might give you the answer the test expects, but there may a lot events before that (probably unnecessarily).